src/fees/hooks.ts

```diff
@@ -32,3 +42,43 @@ export function usePaidFees(fees: Fee[]) {
     totalFee,
   }
 }
+
+// Returns the estimated fee for a SEND or INVITE transaction
+// also optionally fetches new fee estimations if the current ones are missing or out of date
+export function useEstimatedFee(
+  tokenAddress: string,
+  feeType: FeeType.SEND | FeeType.INVITE,
+  shouldRefresh: boolean = true
+) {
+  const dispatch = useDispatch()
+
+  const feeEstimates = useSelector(feeEstimatesSelector)
+
+  // Optionally Keep Fees Up to Date
+  useEffect(() => {
+    if (!shouldRefresh) return
+    const feeEstimate = feeEstimates[tokenAddress]?.[feeType]
+    if (
+      !feeEstimate ||
+      feeEstimate.error ||
+      feeEstimate.lastUpdated < Date.now() - ONE_HOUR_IN_MILLIS
+    ) {
+      dispatch(estimateFee({ feeType, tokenAddress }))
+    }
+  }, [tokenAddress, shouldRefresh])
+
+  const celoAddress = useSelector(celoAddressSelector)
+
+  // feeTokenAddress is undefined if the fee currency is CELO, we still want to
+  // use the fee estimate if that is the case
+  const feeTokenAddress = useFeeCurrency() ?? celoAddress
+
+  const usdFeeEstimate = feeEstimates[tokenAddress]?.[feeType]?.usdFee
+  const feeEstimate =
+    useUsdToTokenAmount(new BigNumber(usdFeeEstimate ?? 0), tokenAddress) ?? new BigNumber(0)
+
+  if (tokenAddress !== feeTokenAddress) {
```

Comment:
This part kind of confuses me. When would this be true? When the token address given isn't a valid fee currency?

Reply:
I guess it's just carried over so hopefully it's fine. might be worth an inline comment though

Reply:
Yea I can add a comment. It happens pretty often that this is true actually.
For instance If a user has 10 cUSD, 1 CELO and tries to withdraw 100% of their cUSD the feeTokenAddress will be chosen as CELO